### ISSUE TITLE
Add validation for configuration parameters

### ISSUE THREAD

**Issue by ryantuck** [fishtown-analytics/dbt#1187]
## config parameter validation

Validation for config parameters would be a useful feature. Sorry if this is a duplicate issue.

### working config parameter as expected

To define a late-binding view in redshift, I can do the following:

```
{{ config(bind=False) }}
select * from public.test
```

Which will yield the following view definition:

```
# \d+ dbt.test_view
                          View "dbt.test_view"
 Column | Type | Collation | Nullable | Default | Storage | Description
--------+------+-----------+----------+---------+---------+-------------
View definition:

  create view "dbt"."test_view__dbt_tmp" as (

select * from public.test
  ) with no schema binding;
```

### silent failure on bad config parameter

However, if I pass in another value that I think will work (like `"false"`) or garbage (like `"asdf"`), `dbt` does not throw any sort of error, and ignores the config input:

```
$ cat test_view.sql
{{ config(bind="false") }}
select * from public.test

$ dbt run --models test
Found 82 models, 19 tests, 0 archives, 0 analyses, 96 macros, 2 operations, 0 seed files

07:45:21 | Concurrency: 8 threads (target='redshift')
07:45:21 |
07:45:21 | 1 of 1 START view model dbt.test_view................................ [RUN]
07:45:22 | 1 of 1 OK created view model dbt.test_view........................... [CREATE VIEW in 1.03s]
07:45:22 |
07:45:22 | Finished running 1 view models in 2.04s.

Completed successfully

Done. PASS=1 ERROR=0 SKIP=0 TOTAL=1
```

The resulting view definition is not 'what I wanted', but `dbt` didn't tell me I had an invalid config value for `bind`:

```
# \d+ dbt.test_view
                           View "dbt.test_view"
 Column |  Type   | Collation | Nullable | Default | Storage | Description
--------+---------+-----------+----------+---------+---------+-------------
 id     | integer |           |          |         | plain   |
View definition:
 SELECT test.id
   FROM test;
```

### who will benefit?

This will help avoid silent failures due to lack of operator knowledge of what exact parameter values are expected.

**Comment by drewbanin** [comment]
Thanks for the report @ryantuck. In the case of `bind` in particular, dbt's implementation of the [create view macro](https://github.com/fishtown-analytics/dbt/blob/dev/grace-kelly/dbt/include/global_project/macros/adapters/redshift.sql#L52) doesn't explicitly check for true/false values. Very unfortunately, the value `"false"` is... truthy... as it is a string of nonzero length.

We can fix this particular issue for `bind` (and other boolean configs like it), but I agree that general validation would also be really great. I think this existed at one point, but when we migrated from Voluptuous to JSON Schemas, that code was removed.
